tests/smoke/features/step_definitions/app.files.ts

```diff
@@ -199,3 +199,100 @@ When(
     }
   }
 )
+
+When(
+  '{string} creates new versions of the folowing file(s)',
+  async function (this: World, stepUser: string, stepTable: DataTable): Promise<void> {
+    const actor = this.actorContinent.get({ id: stepUser })
+    const { allFiles: allFilesPage } = new FilesPage({ actor })
+    const uploadInfo = stepTable.hashes().reduce((acc, stepRow) => {
+      const { to, resource } = stepRow
+
+      if (!acc[to]) {
+        acc[to] = []
+      }
+
+      acc[to].push(this.fileContinent.get({ name: resource }))
+
+      return acc
+    }, {})
+
+    for (const folder of Object.keys(uploadInfo)) {
+      await allFilesPage.uploadFiles({ folder, files: uploadInfo[folder], newVersion: true })
+    }
+  }
+)
+
+When(
+  '{string} declines following resource(s)',
+  async function (this: World, stepUser: string, stepTable: DataTable) {
+    const actor = this.actorContinent.get({ id: stepUser })
+    const { sharedWithMe: sharedWithMePage } = new FilesPage({ actor })
+    const shares = stepTable.raw().map((f) => f[0])
+
+    for (const share of shares) {
+      await sharedWithMePage.declineShare({ name: share })
+    }
+  }
+)
+
+Then(
+  '{string} ensures that the following resource(s) exist',
+  async function (this: World, stepUser: string, stepTable: DataTable) {
+    const actor = this.actorContinent.get({ id: stepUser })
+    const { allFiles: allFilesPage } = new FilesPage({ actor })
+    const resources = stepTable.raw().map((f) => f[0])
+
+    for (const resource of resources) {
+      if (!(await allFilesPage.resourceExist({ name: resource }))) {
+        throw new Error(`resource wasn't find: "${resource}"`)
+      }
+      await allFilesPage.navigate()
+    }
+  }
+)
+
+Then(
+  '{string} ensures that the following resource(s) do(es) not exist',
+  async function (this: World, stepUser: string, stepTable: DataTable) {
+    const actor = this.actorContinent.get({ id: stepUser })
+    const { allFiles: allFilesPage } = new FilesPage({ actor })
+    const resources = stepTable.raw().map((f) => f[0])
+
+    for (const resource of resources) {
+      if (await allFilesPage.resourceExist({ name: resource })) {
+        throw new Error(`resource was find: "${resource}"`)
+      }
+      await allFilesPage.navigate()
+    }
+  }
+)
+
+Then(
```

Comment:
Seems duplicate code to line 255-269? If so, please remove

Reply:
No no, that's right. Differents methods

Reply:
I could swear there were duplicate methods in here 😅 but can't see them anymore